### ISSUE TITLE
segments in wimpsimreader.py (Trac #2161)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2153]
Both segments at [http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/wimpsim-reader/trunk/python/wimpsimreader.py] are supposed to launch I3WimpSimReader module, but they are set with non acceptable values of variable InjectionRadius. The real default is NaN, and not 0 as visible here [http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/wimpsim-reader/trunk/private/wimpsim-reader/I3WimpSimReader.cxx]. The variable has this function (from docs):

''InjectionRadius [Default=NAN] If >0, events will be injected in cylinder with specified radius and [zmin, zmax] height instead of rectangular box''

If the original intention was to avoid the cylinder, this value should be left to NaN, as default. (Or the value 0 should be accepted in I3 module at lines 180-181)

For WimpSimReaderEarth there is an issue with GCD, the code isn't able to read time in given format:

```No registered converter was able to produce a C++ rvalue of type std::string from this Python object of type I3Time```

To avoid it, time should be taken as julian day (or another "good" format), so for example

```line 55: return frame.Get("I3DetectorStatus").start_time```

could be:

```line 55: return frame.Get("I3DetectorStatus").start_time.mod_julian_day_double```

Migrated from https://code.icecube.wisc.edu/ticket/2161
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:15:23", 
    "description": "Both segments at [http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/wimpsim-reader/trunk/python/wimpsimreader.py] are supposed to launch I3WimpSimReader module, but they are set with non acceptable values of variable InjectionRadius. The real default is NaN, and not 0 as visible here [http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/wimpsim-reader/trunk/private/wimpsim-reader/I3WimpSimReader.cxx]. The variable has this function (from docs):\n\n''InjectionRadius [Default=NAN] If >0, events will be injected in cylinder with specified radius and [zmin, zmax] height instead of rectangular box''\n\nIf the original intention was to avoid the cylinder, this value should be left to NaN, as default. (Or the value 0 should be accepted in I3 module at lines 180-181)\n\nFor WimpSimReaderEarth there is an issue with GCD, the code isn't able to read time in given format:\n\n{{{No registered converter was able to produce a C++ rvalue of type std::string from this Python object of type I3Time}}}\n\nTo avoid it, time should be taken as julian day (or another \"good\" format), so for example\n\n{{{line 55: return frame.Get(\"I3DetectorStatus\").start_time}}}\n\ncould be:\n\n{{{line 55: return frame.Get(\"I3DetectorStatus\").start_time.mod_julian_day_double}}}", 
    "reporter": "grenzi", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067323910946", 
    "component": "combo simulation", 
    "summary": "segments in wimpsimreader.py", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2018-06-12T10:09:52", 
    "milestone": "", 
    "owner": "nega", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20180922T09:31:38: olivas changed milestone from "" to "Winter Solstice 2018"

**Comment by icecube-trac** [comment]
Trac update at 20181207T19:34:47: 
* mjl5147 changed status from "new" to "closed"
* mjl5147 changed resolution from "" to "fixed"
* mjl5147 commented:

> In changeset 167396:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="167396"
> Fixing the wimpsim tray segment. First part was already fixed in a previous update. Closes #2153
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190112T04:09:40: nwhitehorn commented:

> In changeset 170619:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170619"
> Fixing the wimpsim tray segment. First part was already fixed in a previous update. Closes #2153
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:15:23: 
* olivas commented:

> Milestone Winter Solstice 2018 deleted


* olivas changed milestone from "Winter Solstice 2018" to ""
